monai/utils/misc.py

```diff
@@ -82,7 +82,20 @@ def star_zip_with(op, *vals):
     return zip_with(op, *vals, mapfunc=itertools.starmap)
 
 
-def first(iterable, default=None):
+T = TypeVar("T")
+
+
+@overload
+def first(iterable: Iterable[T], default: T) -> T:
```

Comment:
just a comment that perhaps we should start to put these definitions in a separate `.pyi` stub file if it doesn't take a lot of effort to maintain...